packages/react-native/ReactCommon/react/performance/timeline/CircularBuffer.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+/**
+ * A container for storing entries of type T, with the following properties:
+ * - It can only grow up to a specified max size
+ * - It's a circular buffer (the oldest elements are dropped if reached max
+ * size and adding a new element)
+ *
+ * Note that the space for maxSize elements is reserved on construction. This
+ * ensures that pointers to elements remain stable across add() operations.
+ */
+template <class T>
+class CircularBuffer {
+ public:
+  explicit CircularBuffer(size_t maxSize) : maxSize_(maxSize) {
+    entries_.reserve(maxSize_);
```

Comment:
If we're already reserving the initial size at the beginning, what's the benefit of using `vector` here over just an array? Maybe we should use an array instead.